components/gateway/pkg/proxy/adaptertransport.go

```diff
@@ -8,87 +8,192 @@ import (
 	"net/http"
 	"strings"
 
+	"github.com/google/uuid"
+
 	"github.com/form3tech-oss/jwt-go"
 	"github.com/kyma-incubator/compass/components/director/pkg/correlation"
 	"github.com/kyma-incubator/compass/components/director/pkg/log"
 	"github.com/kyma-incubator/compass/components/gateway/pkg/httpcommon"
 	"github.com/pkg/errors"
 )
 
+// AdapterConfig stores the configuration for the adapter transport
+type AdapterConfig struct {
+	MsgBodySizeLimit int
+}
+
 type AdapterTransport struct {
+	config AdapterConfig
 	http.RoundTripper
 	auditlogSink AuditlogService
 	auditlogSvc  PreAuditlogService
 }
 
-func NewAdapterTransport(sink AuditlogService, svc PreAuditlogService, trip RoundTrip) *AdapterTransport {
+func NewAdapterTransport(sink AuditlogService, svc PreAuditlogService, trip RoundTrip, config AdapterConfig) *AdapterTransport {
 	return &AdapterTransport{
 		RoundTripper: trip,
 		auditlogSink: sink,
 		auditlogSvc:  svc,
+		config:       config,
 	}
 }
 
+type auditLogContext struct {
+	preAuditLog        PreAuditlogService
+	postAuditLog       AuditlogService
+	cfg                AdapterConfig
+	correlationHeaders correlation.Headers
+	claims             Claims
+	logId              uuid.UUID
+}
+
+func newAuditLogContext(t *AdapterTransport, req *http.Request) (*auditLogContext, error) {
+	correlationHeaders := correlation.HeadersForRequest(req)
+
+	claims, err := getClaims(req.Context(), req.Header)
+	if err != nil {
+		return nil, errors.Wrap(err, "while parsing JWT")
+	}
+
+	transactionId, err := uuid.NewRandom()
+	if nil != err {
```

Comment:
Can we use the style `if x != nil` here and on the other places in the pr? The whole codebase uses this style.

Reply:
I always use nil!=xxx, and it would take me a lot of time to find out all the places I've used it